src/shared/services/auth/services/keycloak.ts

```diff
@@ -0,0 +1,127 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2020 - 2021 Gemeente Amsterdam
+import keycloakJS, { KeycloakInitOptions, KeycloakInstance } from 'keycloak-js'
+import configuration from 'shared/services/configuration/configuration'
+
+const AUTH_REDIRECT_URI = `${global.location.protocol}//${global.location.host}/manage/incidents`
+const SILENT_CHECK_SSO_REDIRECT_URI = `${window.location.origin}/assets/html/silent-check-sso.html`
+const OAUTH_DOMAIN_KEY = 'oauthDomain' // Domain that is used for login
+
+const storage = global.localStorage || global.sessionStorage
+
+class Keycloak {
+  private keycloak: KeycloakInstance
+  private refreshIntervalId: number | null = null
+
+  constructor() {
+    this.keycloak = keycloakJS({
+      clientId: configuration.oidc.clientId,
+      realm: (configuration.oidc as any).realm as string,
+      url: configuration.oidc.authEndpoint,
+    })
+
+    this.keycloak.onAuthSuccess = () => this.startRefreshInterval()
+    this.keycloak.onAuthRefreshError = () => this.stopRefreshInterval()
+    this.keycloak.onAuthError = () => this.stopRefreshInterval()
+    this.keycloak.onAuthLogout = () => this.stopRefreshInterval()
+    this.keycloak.onTokenExpired = () => {
+      // This should never happen (refresh interval should keep token valid)
+      this.stopRefreshInterval()
+      this.logout()
+    }
+  }
+
+  async init() {
+    const options: KeycloakInitOptions = {
+      flow: 'standard', // Standard = code flow
+      checkLoginIframe: false, // To keep user logged in, use refresh token instead of (silent) redirect
+      pkceMethod: 'S256',
+      useNonce: true,
+      silentCheckSsoRedirectUri: SILENT_CHECK_SSO_REDIRECT_URI,
+      silentCheckSsoFallback: true,
+      redirectUri: AUTH_REDIRECT_URI,
+    }
+
+    /**
+     *  If the oauth domain key is set to 'keycloak', this device logged in previously and SSO should be checked.
+     *  For browsers that do not support silent checking (Safari) a redirect will be preformed instead.
+     */
+    if (storage.getItem(OAUTH_DOMAIN_KEY) === 'keycloak') {
+      options.onLoad = 'check-sso'
+    }
+
+    return this.keycloak.init(options)
+  }
+
+  async authenticate() {
+    await this.init()
+
+    if (this.getIsAuthenticated()) {
+      const accessToken = this.getAccessToken()
+
+      return accessToken ? { accessToken } : null
+    }
+
+    return null
+  }
+
+  getIsAuthenticated(): boolean {
+    return Boolean(
+      this.keycloak.authenticated && !this.keycloak.isTokenExpired()
+    )
+  }
+
+  getAuthHeaders(): {
+    Authorization?: string
+  } {
+    const token = this.getAccessToken()
+
+    return token ? { Authorization: `Bearer ${token}` } : {}
+  }
+
+  getAccessToken(): string | null {
+    return this.keycloak.token || null
+  }
+
+  login() {
+    if (typeof global.Storage === 'undefined') {
```

Comment:
When this condition evaluates to true, line 10 could already have thrown a `DOMException`.

Reply:
Removed this check